.travis.yml

```diff
@@ -3,36 +3,59 @@ sudo: false
 addons:
   apt:
     packages:
-    - libcurl4-openssl-dev
-    - libelf-dev
-    - libdw-dev
+      - libcurl4-openssl-dev
+      - libelf-dev
+      - libdw-dev
 rust:
-- nightly
-- beta
-- stable
+  - nightly
+  - beta
+  - stable
 os:
-- linux
+  - linux
 before_script:
-- |
-  pip install 'travis-cargo<0.2' --user &&
-  export PATH=$HOME/Library/Python/2.7/bin:$HOME/.local/bin:$PATH
+  - |
+    pip install 'travis-cargo<0.2' --user &&
+    export PATH=$HOME/Library/Python/2.7/bin:$HOME/.local/bin:$PATH
 script:
-- |
-  sh ./test-no-skeptic.sh &&
-  travis-cargo --only nightly test -- --features "test nightly" -p gluon compile_test &&
-  cargo build --release &&
-  travis-cargo --only nightly bench &&
-  travis-cargo --only stable doc
-after_success:
-- travis-cargo --only stable doc-upload
+  - echo -en 'travis_fold:start:script.test.gluon_base\\r'
+  - travis-cargo test -- -p gluon_base
+  - echo -en 'travis_fold:end:script.test.gluon_base\\r'
+
+  - echo -en 'travis_fold:start:script.test.gluon_parser\\r'
+  - travis-cargo test -- -p gluon_parser --features test
+  - echo -en 'travis_fold:end:script.test.gluon_parser\\r'
+
+  - echo -en 'travis_fold:start:script.test.gluon_check\\r'
+  - travis-cargo test -- -p gluon_check --features test
+  - echo -en 'travis_fold:end:script.test.gluon_check\\r'
+
+  - echo -en 'travis_fold:start:script.test.gluon_vm\\r'
+  - travis-cargo test -- -p gluon_vm --features test
+  - echo -en 'travis_fold:end:script.test.gluon_vm\\r'
+
+  - echo -en 'travis_fold:start:script.test.gluon\\r'
+  - travis-cargo test -- -p gluon --features test
+  - echo -en 'travis_fold:end:script.test.gluon\\r'
+
+  - echo -en 'travis_fold:start:script.test.compile_test\\r'
+  - travis-cargo --only nightly test -- --features "test nightly" -p gluon compile_test
+  - echo -en 'travis_fold:end:script.test.compile_test\\r'
+
+  - echo -en 'travis_fold:start:script.bench\\r'
+  - travis-cargo --only nightly bench
+  - echo -en 'travis_fold:end:script.bench\\r'
+
+  - echo -en 'travis_fold:start:script.build\\r'
+  - cargo build --release
```

Comment:
`cargo bench` already does `cargo build --release` so move this above `bench` (or it makes not sense for nightly